tests/framework/extensions/kubeapi/workloads/template.go

```diff
@@ -11,16 +11,6 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// NewContainer is a contructor that creates a container for a pod template i.e. corev1.PodTemplateSpec
```

Comment:
What's the point of this file?

Reply:
I've moved everything except this `NewImagePullSecret`, an extension that uses the dynamic client to create an image pull secret for a pod template and the necessary conversions.